source/common/filter/config_discovery_impl.h

```diff
@@ -564,6 +603,9 @@ class TcpListenerFilterConfigProviderManagerImpl
           TcpListenerDynamicFilterConfigProviderImpl> {
 public:
   absl::string_view statPrefix() const override { return "tcp_listener_filter."; }
+
+protected:
+  const std::string getConfigDumpType() const override { return "ecds_filter_tcp_listener"; }
```

Comment:
Suggestion: `statPrefix` seems like it accomplishes the same goal of partitioning configs in the stats hierarchy. Maybe reuse?

Reply:
A couple of issue if we reuse statPrefix as the config dump key:

1) Is that naming confusing?
2) Upstream UpstreamHttpFilterConfigProviderImpl has identical stats prefix with HttpFilterConfigProviderImpl. If we go that route, we need to modify them to avoid one config dump override the other.
3)  stats prefix are named as "http", or "tcp", if we use them as config dump keys, it will cause ECDS config dump spreaded around and mixed with other types since config dumps are dumped in the alphabetic order of the keys.

Reply:
Yeah, filed an issue to track 2: https://github.com/envoyproxy/envoy/issues/24010